python/semantic_kernel/memory/chroma_memory_store.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from logging import Logger
+from typing import TYPE_CHECKING, Callable, List, Optional, Tuple
+
+from numpy import ndarray
+from semantic_kernel.memory.memory_record import MemoryRecord
+from semantic_kernel.memory.memory_store_base import MemoryStoreBase
+from semantic_kernel.memory.storage.chroma_data_store import ChromaDataStore
+from semantic_kernel.utils.null_logger import NullLogger
+
+if TYPE_CHECKING:
+    import chromadb.config
+
+ComputeSimilarityFuncType = Callable[[ndarray, ndarray], ndarray]
+
+
+class ChromaMemoryStore(ChromaDataStore, MemoryStoreBase):
+    def __init__(
+        self,
+        logger: Optional[Logger] = None,
+        compute_similarity_fetch_limit: int = 10,
+        compute_similarity_scores_func: ComputeSimilarityFuncType = None,
+        persist_directory: Optional[str] = None,
+        client_settings: Optional["chromadb.config.Settings"] = None,
+    ) -> None:
+        super().__init__(
+            persist_directory=persist_directory, client_settings=client_settings
+        )
+        self._logger = logger or NullLogger()
+        self._compute_similarity_fetch_limit = compute_similarity_fetch_limit
+        self._compute_similarity_scores_func = compute_similarity_scores_func
+
+    async def get_nearest_matches_async(
+        self,
+        collection: str,
+        embedding: ndarray,
+        limit: int = 1,
+        min_relevance_score: float = 0.7,
+    ) -> List[Tuple[MemoryRecord, float]]:
+        collection = await self.get_collection_async(collection)
+        if collection is None:
+            return []
+
+        # use Chroma nearest neighbor search if no similarity function is provided
+        if self._compute_similarity_scores_func is None:
+            query_results = collection.query(
+                query_embeddings=embedding.tolist(),
+                n_results=limit,
+                include=["embeddings", "metadatas", "documents", "distances"],
```

Comment:
Ah I think I got it. The distance != similarity. 

nearest neighbor is a value of 0, most similar is a value of 1. So the returned memories are in correct order, but the min_relevance_score isn't designed for thresholding 'distance.' 

To be consistent, let's have ChromaMemoryStore custom calculate CosineSimilarity (I believe this is why Langchain does this as well).

Reply:
Yes, that's the reason i mentioned this PR has some issues in PR description 
I'll handle it & documentation this weekend :)

Reply:
@joowon-dm-snu Excited to get this in soon!

Reply:
@alexchaomander ;)